### DISCUSSION TITLE
Handling of whitespace in subtitles

### DISCUSSION THREAD

**Discussion by forthrin** [mpv-player/mpv#14930] [Q&A]
A line containing only spaces will terminate parsing of a single SRT subtitle unit, eg. if "_" is a space character ($20),

```
00:00:00,000 --> 00:00:01,000
Hello
_
World
```

... will render as ...

```
00:00:00,000 --> 00:00:01,000
Hello
```

... whereas an empty line with no whitespace ...

```
00:00:00,000 --> 00:00:01,000
Hello

World
```

... will render as ...

```
00:00:00,000 --> 00:00:01,000
Hello
World
```

1. Is this according to de facto parsing conventions?
2. What would go wrong if a space-only line was just skipped, like an empty line?
3. Which code line makes this decision? Couldn't find an immediate candidate:
  
https://git.ffmpeg.org/gitweb/ffmpeg.git/blob/:/libavformat/srtdec.c

**Comment by na-na-hi** [comment]
> 1. Is this according to de facto parsing conventions?

[SRT subtitle format](https://www.matroska.org/technical/subtitles.html#srt-subtitles) doesn't support multiple line breaks. A "blank line" means the start of a new subtitle. In fact, SubRip treats these two examples identically (stops after reading the blank/space-only line), because it strips whitespace first and then check for emptiness, so a line with only whitespaces is considered blank. 

These examples don't have a defined behavior, and different players will render them differently. See also https://github.com/mpv-player/mpv/issues/14271.

**Comment by forthrin** [comment]
Thanks! Pointing out ...

1. The referenced linked does not define what constitutes a "blank line".
2. `nbsp` (which is universally considered white-space), does -not- constitute a "blank line" (by `ffmpeg`, at least). It can thus be used to create empty lines in the ultimate on-screen subtitles.
